packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_widget_configuration.dart

```diff
@@ -0,0 +1,32 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+
+import 'types.dart';
+
+/// A container object for configuration options when building a widget.
+///
+/// This is intended for use as a parameter in platform interface methods, to
+/// allow adding new configuration options to existing methods.
+@immutable
+class MapWidgetConfiguration {
+  /// Creates a new configuration with all the given settings.
+  const MapWidgetConfiguration({
+    required this.initialCameraPosition,
+    required this.textDirection,
+    this.gestureRecognizers = const <Factory<OneSequenceGestureRecognizer>>{},
+  });
+
+  /// The initial camera position to display.
+  final CameraPosition initialCameraPosition;
```

Comment:
It's sort of a weird case; it's not really `MapConfiguration` because it's *initial* position, whereas all of the other configuration in that object is the state the map should have. To illustrate the problem: what would: `updateMapConfiguration(MapConfiguration(initialCameraPosition: somePosition))` do? Even worse, what would:

```
updateMapConfiguration(MapConfiguration(
  initialCameraPosition: somePosition,
  cameraTargetBounds: someOtherLocation,
))
```

do?

This is a fundamental problem with Widget-based plugins (we have the same problem with webview's initial URL), but at least keeping it in the widget config quarantines the weirdness to being widget state. The way I would think of it is that it's widget configuration in that "initial" is only meaningful in the context of the widget creation.

Reply:
Yeah, the fact that it's only required when the map is created, and then we don't want it ever again, is a very good point!

Reply:
(Then we need to update the camera position through the [CameraUpdate](https://github.com/flutter/plugins/blob/main/packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/camera.dart#L118) class, which is its own little piece of evil :S)